### ISSUE TITLE
Allow local mutation

### ISSUE THREAD

**Issue by user72356** [jonaskello/eslint-plugin-functional#349]
Is there a rule that lets me mutate a local object? Something like:

```js
function test()
{
    let myVar = { hello: "" };
    myVar.hello = "world";
}
```

I feel that maybe ```"functional/no-let": ["error", { "allowLocalMutation": true }]``` should allow the code above?

**Comment by RebeccaStevens** [comment]
What error are you currently getting? Is it for the "no-let" or "immutable-data" rule?

**Comment by user72356** [comment]
I get ```'myVar' is never reassigned. Use 'const' instead.eslint[prefer-const]``` for the let myVar

And a ```Modifying an existing object/array is not allowed.eslint[functional/immutable-data]``` under 2nd line.

**Comment by RebeccaStevens** [comment]
I'd recommended setting the option "ignoreAccessorPattern" for "immutable-data".
Then if you rename `myVar` to something like `mutableMyVar` then "immutable-data" will no longer report an issue.

If you aren't reassigning the variable then you should use `const` instead of `let` for the first line.
